### ISSUE TITLE
Filter by nested arrays

### ISSUE THREAD

**Issue by aageneralov** [Orange-OpenSource/hurl#2263]
### Problem to solve

I have some trouble with filter elements in json array by nested arrays elements
Data:

```
{
  "docs": [
    {
      "title": "foo",
      "authors": [
        "one",
        "two"
      ]
    },
    {
      "title": "bar",
      "authors": [
        "one",
        "three"
      ]
    },
    {
      "title": "baz",
      "authors": [
        "two",
        "three"
      ]
    }
  ]
}  
```
I need to assert that data have doc with author named "two", I tried to use many variants of jsonpath:

``` 
$.docs[?(@.authors.indexOf('two') != -1)]
$.docs[?(@.authors[*] == 'two')]
$.docs[?(@.authors[?(@ == 'two')])]
```

But it doesn't work

### Proposal

Please add an `indexOf` function or something like that so that you can filter data by nested arrays

**Comment by dorryspears** [comment]
@fabricereix what do you think about adding a "for all" clause on the on the assert

So for this example you would write 

jsonpath $..authors contains "two" for all

**Comment by fabricereix** [comment]
yes we plan to add predicates that are applied to each element of a list.
you can have a look at https://github.com/Orange-OpenSource/hurl/issues/347

**Comment by dorryspears** [comment]
@fabricereix I will try to tackle the feature

**Comment by CrispyDrone** [comment]
What do you think about integrating hurl with any arbitrary tools to do processing? So, you could just invoke a `jq` or `yq` command during filters, assertions,...

**Comment by unhammer** [comment]
`indexOf` or similar would be great. I've got a list of strings where I need to check that one of the strings contains a substring. 

(Even a way to concat the list into one string (like an Array.join) would also be a possible workaround, then one could `Join($[*].name, ",") contains "bo"` to check we've got at least one of Bilbo or Rambo in there.)

**Comment by fabricereix** [comment]
Hi @aageneralov,
We have now implemented the official JSONPath spec (#4270).
In the next release, the following jsonpath queries will be supported.
```
$.docs[?(@.authors[1]=="two" || @.authors[0]=="two")].
$.docs[?(@.authors[?(@ == 'two')])]
```

We still plan to implement a special "each" operator ([#347](https://github.com/Orange-OpenSource/hurl/issues/347)).
I'm closing this issue because it is not about jsonpath anymore.
